src/manager.cpp

```diff
@@ -203,12 +203,101 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    nlohmann::json l_jsonObj{};
+    int l_bytesWritten = -1;
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    if (m_worker.get() != nullptr)
+    {
+        l_jsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    try
+    {
+        int l_bytesWrittenOnDbus = -1;
+
+        // For the given VPD path, get it's corresponding inventory object path
+        // from system config JSON
+        const std::string& l_inventoryObjPath =
```

Comment:
Config Json is mandatory to use update keyword API?

Reply:
yes.

In the case where you give me any path, which is not present in JSON, we will end up parsing the given input string to check if it is dbus object path or hardware path. and we discussed not to have such checks as it doesn't gaurantee on what we are trying to achieve

Reply:
to deal with this, we should have two write APIs. one which writes on dbus, provided inventory dbus object path.
another one which writes on hardware, provided hardware path.

Reply:
if we have two specific apis, then even if the given path is not present in JSON, we can confidently perform the respective write operations.

Reply:
As decided, code will update hardware in case no Json found.